lib/order.rb

```diff
@@ -0,0 +1,84 @@
+require 'line_length'
+
+class Order
+  include LineLength
+  attr_reader :ordered_dishes
+
+  def initialize
+    @ordered_dishes = []
+  end
+
+  def add_to_order(dish,quantity)
+    @ordered_dishes << {dish:dish,quantity:quantity}
+  end
+
+  def order_total
+    @order_total = 0
+    ordered_dishes.each{|dish| @order_total+=dish[:dish].price*dish[:quantity]}
+    @order_total
+  end
+
+  def display_order_on_receipt
+    display = display_order_title
+    display += display_order_body
+    display += display_order_total
+  end
+
+  def display_order_on_sms
+    display = shorten_spaces(display_order_on_receipt)
+    display = shorten_line(display)
+  end
+
+  private
+
+  def shorten_spaces(display)
+    display.gsub!("\n\n","\n")
+    display.gsub!(" ","")
+    display.gsub!("£"," £ ")
+    display
+  end
+
+  def shorten_line(display)
+    totals_line = "-"*(line_length)
+    new_totals_line = "-"*(line_length/2)
+    display.gsub!(totals_line,new_totals_line)
+    display
+  end
+
+  def display_order_title
+    title = "Order Details"
+    title_spacing = (line_length - title.length)/2
+    " "*title_spacing + "#{title}\n\n"
+  end
+
+  def display_order_body
+    display = ''
+    ordered_dishes.each{|dish| display += add_line_to_order_display(dish)}
+    display
+  end
+
+  def display_order_total
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for display_order_total is too high. [18.03/15]